cpp/src/groupby/sort/group_tdigest.cu

```diff
@@ -57,7 +57,8 @@ struct make_centroid {
 
   centroid operator() __device__(size_type index)
   {
-    return {static_cast<double>(col.element<T>(index)), 1, col.is_valid(index)};
+    return {
+      static_cast<double>(col.element<T>(index)), col.is_valid(index) ? 1 : 0, col.is_valid(index)};
```

Comment:
So if the row at `index` is null, this will store `mean` as an arbitrary value reading from `col.element<T>(index)`?
```suggestion
    auto const is_valid = col.is_valid(index);
    auto const mean = is_valid ? static_cast<double>(col.element<T>(index)) : 0.0;
    return { mean, static_cast<double>(is_valid), is_valid};
```

Reply:
This seems reasonable.  The value never gets used elsewhere but storing a 0 with your change eliminates a memory read.

Reply:
Note:  I left the weight calculation as a ternary.  I'm not a big fan of implicitly converting bool values to values that just happen to be either 1.0 or 0.0.